cmd/snapshooter/snap.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"time"
+
+	ptime "github.com/mintel/elasticsearch-asg/pkg/time" // Time utilities
+)
+
+// SnapshotWindow represents how often to take Elasticsearch snapshots,
+// and how long to keep them.
+type SnapshotWindow struct {
+	Every   time.Duration // Take snapshots with this frequency.
+	KeepFor time.Duration // Keep snapshots for this long.
+}
+
+// NewSnapshotWindow returns a new SnapshotWindow by parsing two ISO 8601 Duration strings.
+//
+// It returns an error if the duration strings cannot be parsed.
+func NewSnapshotWindow(every, keepFor string) (SnapshotWindow, error) {
+	sw := SnapshotWindow{}
+
+	d, err := ptime.ParseISO8601D(every)
+	if err != nil {
+		return SnapshotWindow{}, err
```

Comment:
Why not return sw since that's already been created?

Reply:
Good point.